crates/store/src/state.rs

```diff
@@ -154,62 +148,70 @@ impl State {
                         error,
                     }
                 })?;
-                if peaks.hash_peaks() != block_header.chain_root() {
+                if peaks.hash_peaks() != block.header().chain_root() {
                     return Err(ApplyBlockError::NewBlockInvalidChainRoot);
                 }
 
-                chain_mmr.add(block_header.hash());
+                chain_mmr.add(block.header().hash());
                 chain_mmr
             };
 
             // update nullifier tree
             let nullifier_tree = {
                 let mut nullifier_tree = inner.nullifier_tree.clone();
-                for nullifier in nullifiers.iter() {
+                for nullifier in block.created_nullifiers() {
                     nullifier_tree
-                        .insert(nullifier, block_header.block_num())
+                        .insert(nullifier, block.header().block_num())
                         .map_err(ApplyBlockError::FailedToUpdateNullifierTree)?;
                 }
 
-                if nullifier_tree.root() != block_header.nullifier_root() {
+                if nullifier_tree.root() != block.header().nullifier_root() {
                     return Err(ApplyBlockError::NewBlockInvalidNullifierRoot);
                 }
                 nullifier_tree
             };
 
             // update account tree
             let mut account_tree = inner.account_tree.clone();
-            for update in &accounts {
+            for update in block.updated_accounts() {
                 account_tree.insert(
-                    LeafIndex::new_max_depth(update.account_id.into()),
-                    update.final_state_hash.into(),
+                    LeafIndex::new_max_depth(update.account_id().into()),
+                    update.new_state_hash().into(),
                 );
             }
 
-            if account_tree.root() != block_header.account_root() {
+            if account_tree.root() != block.header().account_root() {
                 return Err(ApplyBlockError::NewBlockInvalidAccountRoot);
             }
 
-            // build notes tree
-            let note_tree = build_note_tree(&notes)?;
-            if note_tree.root() != block_header.note_root() {
+            // build note tree
+            let note_tree = build_note_tree(block.notes())?;
+            if note_tree.root() != block.header().note_root() {
                 return Err(ApplyBlockError::NewBlockInvalidNoteRoot);
             }
 
-            drop(span);
```

Comment:
Do we no longer need this `drop(span)`?

Reply:
You're right, I removed it during refactoring, but reverted the part after drop, so I should restore this `drop`, thank you!